### PR TITLE
Refactored probability distribution comparisons

### PR DESCRIPTION
**Pull request by MattsonCam**
In this pr, I refactored the comparisons between probability distributions within phenotypes, treatments, treatment types, and plates. Please let me know if you have any questions or suggestions.

### CODE REVIEW

File: 3.analyze_data/utils/significance_testing.py [hunk 1]
```diff
@@ -134,37 +134,52 @@ def get_treatment_comparison(_comp_functions, _treatment_paths, _probadf, _barco
         negcondf = common_broaddf.loc[common_broaddf["control_type"] == "negcon"]
         no_negcondf = common_broaddf.loc[common_broaddf["control_type"] != "negcon"]
 
-        # Create comparison groups
-        iter_group = set(zip(no_negcondf["Metadata_plate"], no_negcondf["Metadata_model_type"], no_negcondf[treat_data["treatment_column"]]))
+        # Create groups for making comparisons using the model type and the treatment
```
Comment (MikeLippincott) [inline 3.analyze_data/utils/significance_testing.py:137]: Great, Documentation!

File: 3.analyze_data/utils/significance_testing.py [hunk 2]
```diff
@@ -134,37 +134,52 @@ def get_treatment_comparison(_comp_functions, _treatment_paths, _probadf, _barco
...
+
+            # Compute the comparisons within each plate seperately
+            for plate in groupdf["Metadata_plate"].unique():
+
+                # Specify the treatment and negative control dataframes
+                treatdf = no_negcondf.loc[(no_negcondf["Metadata_plate"] == plate) & (no_negcondf["Metadata_model_type"] == model_type) & (no_negcondf[treat_data["treatment_column"]] == utreat)]
+                negdf = negcondf.loc[(negcondf["Metadata_plate"] == plate) & (negcondf["Metadata_model_type"] == model_type)]
+
+                # Remove wells if the cell count is below the corresponding threshold
+                treatdf = filter_wells_by_cell_count(treatdf, _treat_cutoff)
```
Comment (MikeLippincott) [inline 3.analyze_data/utils/significance_testing.py:157]: I am curious as to how these cutoffs are defined?

**Review by MikeLippincott — APPROVED** [review]
LGTM. Nice PR.

File: 3.analyze_data/utils/significance_testing.py (hunk 2 above)
Comment (MattsonCam) [reply 3.analyze_data/utils/significance_testing.py:157]: I just choose a constant to define the cutoffs.

**Comment by MattsonCam** [comment]
Thanks @MikeLippincott, merging now!

**Pull request merged**